### PR TITLE
perf(*) reduce unnecessary read on ngx.var and avoid caching outside

### PR DESCRIPTION
**Pull request by fffonion**
of uncommon branches

Other work like reducing var.http_HEADER and upstream var will be done in seperate PR.

Partly from #7426 #7421

### CODE REVIEW

**Review by javierguerragiraldez — APPROVED** [review]

File: kong/runloop/handler.lua [hunk 1]
```diff
@@ -1206,37 +1207,38 @@ return {
...
-        return kong.response.exit(426, { message = "Please use HTTP2 protocol" }, {
-          ["connection"] = "Upgrade",
-          ["upgrade"]    = "HTTP/2",
-        })
-      end
+      if protocols.grpc or protocols.grpcs then
+        -- perf: branch usually not taken, don't cache var outside
+        local content_type = var.content_type
+
+        if content_type and find(content_type, "application/grpc", 1, true) == 1 then
```
Comment (dndx) [inline kong/runloop/handler.lua:1219]: I think the original version of using `sub(content_type, 1, #"application/grpc") == "application/grpc"` would be faster than `string.find`.

Comment (fffonion) [reply kong/runloop/handler.lua:1219]: I did a quick test, sub is indeed (slightly) faster than find but are in the noise margin. Consider this branch
is only taken in grpc, i will revert it back to sub to reduce diff from original.

Comment (bungle) [reply kong/runloop/handler.lua]: perhaps "slower" but does the sub generate more garbage?

Comment (javierguerragiraldez) [reply kong/runloop/handler.lua]: no, there's some escape analysis and operation folding.  if JITted, the substring is never created, the comparison is made in-place.

**Pull request merged**